src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/ActionButtons.tsx

```diff
@@ -62,20 +70,32 @@ const ActionButtons = ({
 
   return (
     <div className='discovery-modal_buttons-row' data-testid='actionButtons'>
-      <DownloadAllModal
-        downloadStatus={downloadStatus}
-        setDownloadStatus={setDownloadStatus}
+      <DownloadModal
+        downloadStatus={downloadAllStatus}
+        setDownloadStatus={setDownloadAllStatus}
+      />
+      <DownloadModal
+        downloadStatus={downloadVariableMetadataStatus}
+        setDownloadStatus={setDownloadVariableMetadataStatus}
       />
       <Row className='row'>
-        {/*
-        <Col flex='1 0 auto'>
-          <Button className='discovery-action-bar-button'>
-            Download <br />
-            Variable-Level Metadata
-          </Button>
-        </Col>
-        */}
-        {showDownloadStudyLevelMetadataButtons && (
+        {showDownloadVariableMetadataButton && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => {
+                DownloadVariableMetadata(
+                  resourceInfo,
+                  setDownloadVariableMetadataStatus,
```

Comment:
My concern is that if they both share the same state, unexpected UI behavior could occur that would make it difficult to use the application in some edge cases. For example, a user could press the _Download All Files_ button, and while that is downloading they could press the _Download Variable Metadata_ button. I think it would be  possible that they could both fail one after the other, and the application would hide/show and change the text of the modal while the user is reading it if they share the same state. 

One alternative to hard coding individual Download Modal components would be to dynamically generate them by storing all the Download Modal statuses in a state containing an array of these statuses _DownloadModalStatuses_. Then we could create each DownloadModal by mapping through the DownloadModalStatuses. 

This might be a good idea if there are going to be many download statuses. Let me know what you think.

Reply:
Yeah the edge case you mentioned makes sense. I think the regular way we were using to deal with this situation is to let only one button in this group being able to get triggered at any certain time. That is, when user clicks on a button, that button goes into `pending` + `disabled` state, while all the other buttons will go into `disable` state. And they don't get re-enabled until the pervious button action has finished. This can be done bu using the shared state variable for download status across this button group.
We can discuss further about this approach if you would like.

Reply:
I couldn't find a pending state in AntD's documentation, but I implemented this using a loading state. The _inProgress_ data type in the _downloadStatus_ is now a string and either references what is currently in progress or is an empty string. The Download All Files button and the Download Variable-level Metadata disable all buttons and display the loading icon while downloading and then reset after. The other two buttons do not, as they download instantaneously.